### PR TITLE
[WIP] JS Extender API foundation

### PR DESCRIPTION
**Pull request by tobscure**
Kicking off #1317

### CODE REVIEW

**Comment by franzliedke** [comment]
This puts the infrastructure in place for running an extension's JS extender, right? :+1: So, can we merge this, to keep future PRs small? :)

**Comment by tobscure** [comment]
Yes, just a couple more minor changes I need to push in order to ensure that these extenders will actually work.

**Comment by datitisev** [comment]
Question about importing lodash-es: Webpack will only import the export flattenDeep, and not the whole npm module, into the dist file, right?
Just wanna make sure we aren’t adding a lot of unused JS code 🙂.
I forgot so, one can just get the code from https://30secondsofcode.org

**Comment by tobscure** [comment]
Correct! Webpack does tree-shaking automatically.

File: js/src/common/extend/Model.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+export default class Routes {
```
Comment (franzliedke) [inline js/src/common/extend/Model.js]: That should be `Model`, I suppose. ;)

**Pull request merged**